src/Microsoft.CorrelationVector/CorrelationVectorV1.cs

```diff
@@ -0,0 +1,404 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Globalization;
+
+namespace Microsoft.CorrelationVector
+{
+    /// <summary>
+    /// This class represents a lightweight vector for identifying and measuring
+    /// causality.
+    /// </summary>
+    public sealed class CorrelationVectorV1 : CorrelationVector
+    {
+        internal new const byte MaxVectorLength = 63;
+        internal const byte MaxVectorLengthV2 = 127;
+        internal const byte BaseLength = 16;
+        internal const byte BaseLengthV2 = 22;
```

Comment:
We don't need both the V1 and V2 properties here. Let's split them out into their respective classes